### PR TITLE
Qt5 connection with slot as method argument

### PR DESCRIPTION
**Pull request by 3nids**
also autosipify the file

## Checklist

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: python/gui/symbology-ng/qgsrendererpropertiesdialog.sip [hunk 1]
```diff
@@ -1,93 +1,141 @@
...
-     * Connect the given slot to the value changed event for the set of widgets
-     * Each widget is checked for type and the common type of signal is connected
-     * to the slot.
-     *
-     * @param widgets The list of widgets to check.
-     * @param slot The slot to connect to the signals.
-     */
-    void connectValueChanged( const QList<QWidget *>& widgets, const char *slot );
 
+    void connectValueChanged( const QList<QWidget *> &widgets, void ( QgsRendererPropertiesDialog::*slot )() );
```
Comment (3nids) [inline python/gui/symbology-ng/qgsrendererpropertiesdialog.sip]: this does not build in python. Any hint?

Comment (m-kuhn) [inline python/gui/symbology-ng/qgsrendererpropertiesdialog.sip]: `const char *slot`?

Comment (3nids) [inline python/gui/symbology-ng/qgsrendererpropertiesdialog.sip]: thanks, it compiltes, will try it out.

Comment (3nids) [inline python/gui/symbology-ng/qgsrendererpropertiesdialog.sip]: image

Comment (nyalldawson) [inline python/gui/symbology-ng/qgsrendererpropertiesdialog.sip]: Does it need to be in Python? I don't think it's required

Comment (3nids) [inline python/gui/symbology-ng/qgsrendererpropertiesdialog.sip]: well, it not does it need to be a dedicated method either? It seems to be only called once from the same class.

Comment (3nids) [inline python/gui/symbology-ng/qgsrendererpropertiesdialog.sip]: anyway, I would still be interested on how to solve this

Comment (3nids) [inline python/gui/symbology-ng/qgsrendererpropertiesdialog.sip]: same question will still be valid for `QgsAnimatedIcon::connectFrameChanged`

Comment (3nids) [inline python/gui/symbology-ng/qgsrendererpropertiesdialog.sip]: some hints maybe [here](https://github.com/baoboa/pyqt5/blob/master/sip/QtQml/qqmlproperty.sip#L79-L92) and in [extension API](http://pyqt.sourceforge.net/Docs/PyQt5/extension_api.html)

**Comment by m-kuhn** [comment]
Possibilities:

 * Add a "classic" method in the .h that takes a `const char *`, expose this to python (see QgsAnimatedIcon)
 * Write some %MethodCode that does something similar to [fromSignal](http://doc.qt.io/qt-5/qmetamethod.html#fromSignal) but it needs to be `fromSlot` (iterate over available metamethods and compare them with the provided function pointer to find the correct one). Extra points to create a pull request for that upstream at Qt
 * Get new style signal support upstream in pyqt5
 * Don't streamline this, and force devs to do manual work for it. Justify it with the approach often being a "bad design decision".

**Comment by 3nids** [comment]
I gave it a try yesterday with some MethodCode without much success.

**Comment by m-kuhn** [comment]
AFAICT `QgsRendererPropertiesDialog` is not subclassed, so go for option number 4 and make it private?

**Comment by 3nids** [comment]
yep, of course. My intention was to use this for QgsAnimatedIcon too, indeed here it doesn't make much sense.

**Comment by 3nids** [comment]
thanks for the hints anymway

**Pull request closed without merging**